v1.2 docs update (#404)
flarum/docs: docs/extend/update-1_x.md

:::

## 1.2 Changes

Flarum v1.2 included quite a few bugfixes, internal refactors, and new features. The following recaps the most important changes for extension developers:

### Frontend

> - Flarum core now passes TypeScript type checking (on the portion written in TypeScript). Additionally, major portions of the frontend (models, the application instance, and others) are now written in TypeScript. These changes should make it much easier and more fruitful to write extensions in TypeScript.

Reviewer: Should there be blank lines between each of these paragraphs?

> - Instead of directly using LESS variables in CSS code, core now uses CSS variables. For the most part, we've just created CSS variables and set their values to the LESS variables. This should make theming and customizing CSS a lot easier. https://github.com/flarum/core/pull/3146.

Reviewer: ```suggestion
- Instead of directly using Less variables in CSS code, core now uses CSS variables. For the most part, we've just created CSS variables and set their values to the Less variables. This should make theming and customizing CSS a lot easier. https://github.com/flarum/core/pull/3146.
```

- Dropdowns can now be lazy-drawn to improve performance. You can do this by setting the lazy draw attr to "true". https://github.com/flarum/core/pull/2925.
- [Textarea-type settings](https://github.com/flarum/core/pull/3141) are now supported through the `app.extensionData.registerSetting` util.

> - You can now use webpack v5 to bundle your extension's code.

Reviewer: ```suggestion
- You can now use Webpack 5 to bundle your extension's code. This will offer minor bundle size improvements.
```

- A new `flarum/common/components/ColorPreviewInput` component [has been added](https://github.com/flarum/core/pull/3140). It can be used directly, or through the `color-preview` type when registered via `app.extensionData.registerSetting`.
- Extensions can now [modify the minimum search length](https://github.com/flarum/core/pull/3130) of the `Search` component.
- The following components are now extensible:

>   - The "colors" part of the Appearance page in admin: https://github.com/flarum/core/pull/3186

Reviewer: ```suggestion
  - The "colors" part of the Appearance page in the admin dashboard: https://github.com/flarum/core/pull/3186
```

  - The `StatusWidget` dropdown in the admin dashboard: https://github.com/flarum/core/pull/3189
  - `primaryControls` in the notification list: https://github.com/flarum/core/pull/3204
- Extensions now have finer control over positioning when adding elements to the DiscussionPage sidebar items: https://github.com/flarum/core/pull/3165
- 

### Backend

- An [extender for settings defaults](https://github.com/flarum/core/pull/3127) has been added. This should be used instead of the `addSettings` migration helper, which has been deprecated.

> - You can now [generate LESS variables from setting values](https://github.com/flarum/core/pull/3011) via the `Settings` extender.

Reviewer: ```suggestion
- You can now [generate Less variables from setting values](https://github.com/flarum/core/pull/3011) via the `Settings` extender.
```

- Extensions can now [override/supplement blade template namespaces](https://github.com/flarum/core/pull/3167) through the `View` extender, meaning custom blade templates can be used instead of the default ones added by core or extensions.

> - You can now define [custom LESS functions through PHP](https://github.com/flarum/core/pull/3190), allowing you to use some backend logic in your LESS theming.

Reviewer: ```suggestion
- You can now define [custom Less functions through PHP](https://github.com/flarum/core/pull/3190), allowing you to use some backend logic in your Less theming.
```

### Backend

- Flarum now comes with a [Preload extender](https://github.com/flarum/core/pull/3057) for preloading any custom frontend assets.

> - A new [Theme](https://github.com/flarum/core/pull/3008) extender now allows overriding Less files and internal imports. This allows themes to more easily completely replace Less modules.

Reviewer 2: Not sure what changed on this line. EOF?
